Quotes: JS refactor

bleistivt opened the pull request:
* Remove livequery dependency.
* Remove the editor stack concept and insert quotes in the last TextBox that had focus instead.
* Pull the function definitions out of the constructor.
* Make quote folding strings translateable.
* Remove commented out code.
* Remove dead code.
* Added comments.
* Lord Brackos


Tested with: ButtonBar, editor.

linc commented:
In general, try to do code format refactors as a separate PR before changing all the functionality. It gets really hard to read the diff when you do both together. We'll muddle thru for this one. 

I need to check internally here to see if any clients are still stuck on cleditor + quotes before we merge this. Hopefully we can expedite them getting off of it.

bleistivt commented:
Yes, this commit became larger and larger as i moved on :laughing: 

I removed cleditor because it was in the archive section of vanilla/addons.
I could add it back in for the time it is still being used if you want.

linc commented:
If you want to add it back, that'd definitely clear us to merge this quicker. Just clearly document what we can rip back out once the coast is clear. I bet there are open source folks still using somewhere too.
